### ISSUE TITLE
Linker Support

### ISSUE THREAD

**Issue by aritchie** [shinyorg/shiny#171]
# IF YOU DON'T ANSWER THIS TEMPLATE - THE BOT WILL AUTOMATICALLY CLOSE YOUR ISSUE! 

## Summary
Shiny is currently not linker friendly

## API Changes
None 

## Intended Use Case
Linker is nice for shrinking app size

## Notes
Related to #160 
In hanselman.forms I am trying to turn Link All on.

Getting: System.InvalidOperationException: 'A suitable constructor for type 'Shiny.Jobs.JobManager' could not be located. Ensure the type is concrete and services are registered for all parameters of a public constructor.'

Had to add Shiny.Core to the list to skip in linking. Could probably throw a preserve on there to make sure it works properly.

Additionally, I had to add a [Preserve(AllMembers = true)] to my IJob implementation

**Comment by aritchie** [comment]
Currently blocked by #190

**Comment by michalhoblik** [comment]
Hey @aritchie. 

First of all, thank you for this great library. Are you planning to support linking for Shiny in the near future? Currently we have 2 apps with where we have just Shiny.Core via Shiny.Prism package integrated. We are unable to link both the apps on Android even if we add Shiny.Core to the list to skip in linking process. Because of this our android apps with not so many features inside have size more >80MB. We also tried to add Shiny.Prism and Shiny.Core to the list, nothing helps. Are we missing something in the configuration oder someting has to be done in Shiny.Core? Thanks in advance for any information.

Michal

**Comment by aritchie** [comment]
@michalhoblik this is not a priority at all for me right now.
